src/main/java/wooteco/subway/admin/service/LineService.java

```diff
@@ -29,26 +33,47 @@ public Line save(Line line) {
         return lineRepository.findAll();
     }
 
-    public void updateLine(Long id, Line line) {
+    public Line updateLine(Long id, Line line) {
         Line persistLine = lineRepository.findById(id).orElseThrow(RuntimeException::new);
         persistLine.update(line);
-        lineRepository.save(persistLine);
+        return lineRepository.save(persistLine);
     }
 
     public void deleteLineById(Long id) {
         lineRepository.deleteById(id);
     }
 
-    public void addLineStation(Long id, LineStationCreateRequest request) {
-        // TODO: 구현
+    public void addLineStation(Long id, LineStation lineStation) {
+        Line line = lineRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException(NO_SUCH_LINE));
+        line.addLineStation(lineStation);
+        lineRepository.save(line);
     }
 
     public void removeLineStation(Long lineId, Long stationId) {
-        // TODO: 구현
+        Line line = lineRepository.findById(lineId)
+                .orElseThrow(() -> new NoSuchElementException(NO_SUCH_LINE));
+        line.removeLineStationById(stationId);
+        lineRepository.save(line);
     }
 
     public LineResponse findLineWithStationsById(Long id) {
-        // TODO: 구현
-        return new LineResponse();
+        Line line = lineRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException(NO_SUCH_LINE));
+        return LineResponse.of(line, findStationsOf(line));
+    }
+
+    public Set<Station> findStationsOf(Long id) {
+        Line line = lineRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException(NO_SUCH_LINE));
+        return findStationsOf(line);
+    }
+
+    private Set<Station> findStationsOf(Line line) {
+        Set<Station> stations = new HashSet<>();
+        for (Station station : stationRepository.findAllById(line.getLineStationsId())) {
+            stations.add(station);
+        }
+        return stations;
```

Comment:
```suggestion
        return stationRepository.findAllById(line.getLineStationsId());
```

`StationRepository`에서 `findAllById`를 오버라이드하여 위와 같은 코드로 만들 수 있습니다.

Reply:
`stationRepository.findAllById()`가 `line.getStationIds()`의 순서를 보장하지 않아서 다음과 같이 변경했습니다.

```java
List<Station> stations = new ArrayList<>();
    for (Long stationId : line.getStationIds()) {
        stationRepository.findById(stationId)
            .ifPresent(stations::add);
}
return stations;
```

이와 같은 경우도 혹시 오버라이드를 활용해 순서를 보장할 수 있을까요?

Reply:
반복문 내에 쿼리를 조회하는 코드가 있다면 성능상 큰 문제가 발생할 수 있습니다.
만약 100개의 Line이 있다면 100개의 쿼리가 발생하겠네요..!!
순서의 보장을 위해서는 ORDER BY를 활용하거나, WHERE IN을 사용하여 조회 후 정렬하는 로직이 필요할 것 같아요.